monkey/infection_monkey/utils/plugins/plugin.py

```diff
@@ -47,13 +47,29 @@ def get_instances(cls) -> Sequence[Type[Plugin_type]]:
                 LOG.debug("Checking if should run object {}".format(class_object.__name__))
                 try:
                     if class_object.should_run(class_object.__name__):
-                        instance = class_object()
-                        objects.append(instance)
+                        objects.append(class_object)
                         LOG.debug("Added {} to list".format(class_object.__name__))
                 except Exception as e:
                     LOG.warning("Exception {} when checking if {} should run".format(str(e), class_object.__name__))
         return objects
 
+    @classmethod
+    def get_instances(cls) -> Sequence[Type[Plugin_type]]:
```

Comment:
I've probably missed it before, but why is `Plugin_type` not `PluginType`?

Reply:
Can't return `Plugin` because it's not defined yet when the type checker reaches that line (we're still defining Plugin at that point).
Forward references in typing are I think only in 3.8.
So messing around with this https://mypy.readthedocs.io/en/latest/generics.html

Reply:
Ah, if you mean PEP reasons, no particular reason.